unity/Holo/Assets/SharedExperience/SharingSceneData.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+using UnityEngine.Networking.Match;
+
+public class SharingSceneData : NetworkBehaviour
+{
+    [SyncVar]
+    int hostInstanceIndex;
+
+    [SyncVar]
+    Vector3 hostPlatePosition;
+
+    [SyncVar]
+    Vector3 hostPlateScale;
+
+    [SyncVar]
+    Quaternion hostModelRotation;
+
+    [SyncVar]
+    bool hostClippingPlaneActive;
+
+    [SyncVar]
+    Transform hostClippingPlaneTransform;
+
+    [SyncVar]
+    string hostColorMap;
+
+    [SyncVar]
+    bool hostAnimationState;
+
+    [SyncVar]
+    float hostAnimationTime; // maybe we should set it only when pause or any state change
+
+    [SyncVar]
+    float hostAnimationSpeed;
+
+    //temporary solution for dataflow
+    [SyncVar]
+    bool hostDataLayerActive;
+
+
+    ModelWithPlate ModelManager;
+    ModelClippingPlaneControl ClipPlaneManager;
+    ColorMap ColorMapManager;
+
+    void Start()
+    {
+        ModelManager = gameObject.GetComponent<ModelWithPlate>();
+        ClipPlaneManager = ModelManager.ModelClipPlane.GetComponent<ModelClippingPlaneControl>();
+        ColorMapManager = gameObject.GetComponent<ColorMap>();
+
+        hostInstanceIndex = (ModelManager.instanceIndex.HasValue) ? ModelManager.instanceIndex.Value : -1;
+        hostPlatePosition = transform.localPosition;
+        hostPlateScale = transform.localScale;
+
+        hostClippingPlaneActive = ClipPlaneManager.ClippingPlaneState != ModelClippingPlaneControl.ClipPlaneState.Disabled;
+        hostClippingPlaneTransform = ModelManager.ModelClipPlane.transform;
+
+        hostColorMap = ModelManager.DataVisualizationMaterial.GetTexture("_ColorMap").name;
```

Comment:
Instead of accessing `ModelManager.DataVisualizationMaterial.GetTexture("_ColorMap").name`, we should rather use `GetComponent<ColorMap>().ColorMap`, and expose in `ColorMap.cs` the property `ColorMap` string. A setter for this property can be extracted from current `ColorMap.ClickSetColorMap` method.